File: app/src/main/java/com/example/amazonprimeclone/fragments/home/series/SeriesFragmentViewModel.java
package com.example.amazonprimeclone.fragments.home.series;

import static com.example.amazonprimeclone.data.remote.Constants.API_KEY;

import androidx.annotation.NonNull;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.amazonprimeclone.modal.SearchSeries;
import com.example.amazonprimeclone.modal.SeriesCategory;
import com.example.amazonprimeclone.modal.SeriesCategoryResults;
import com.example.amazonprimeclone.modal.SeriesResponseResults;
import com.example.amazonprimeclone.data.remote.Constants;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class SeriesFragmentViewModel extends ViewModel {

    private final MutableLiveData<List<SeriesResponseResults>> slideList;
    private final MutableLiveData<List<SeriesCategoryResults>> resultsList;

    public SeriesFragmentViewModel(){
        slideList = new MutableLiveData<>();
        resultsList = new MutableLiveData<>();
        getData();
    }

    public MutableLiveData<List<SeriesCategoryResults>> getResultsList() {
        return resultsList;
    }

    public MutableLiveData<List<SeriesResponseResults>> getSlideList() {
        return slideList;
    }

    private void getData() {
        loadSlideList();
        loadResultList();
    }

    private void loadSlideList() {
        Call<SearchSeries> movieResponseCall = Constants.retrofitService.getTopRatedSeries();
        movieResponseCall.enqueue(new Callback<SearchSeries>() {
            @Override
            public void onResponse(@NonNull Call<SearchSeries> call, @NonNull Response<SearchSeries> response) {
                SearchSeries category = response.body();
                //add to list
                if(category!=null){
                    slideList.setValue(category.getResults().subList(0,9));
                }

            }

            @Override
            public void onFailure(@NonNull Call<SearchSeries> call, @NonNull Throwable t) {

            }
        });

    }

    private void loadResultList() {
        Call<SeriesCategory> movieResponseCall = Constants.retrofitService.getCategoriesSeries(API_KEY);
        movieResponseCall.enqueue(new Callback<SeriesCategory>() {
            @Override
            public void onResponse(@NonNull Call<SeriesCategory> call, @NonNull Response<SeriesCategory> response) {
                SeriesCategory category = response.body();
                //add to list
                if(category!=null){
                    resultsList.setValue(category.getGenres());
                }

            }

            @Override
            public void onFailure(@NonNull Call<SeriesCategory> call, @NonNull Throwable t) {

            }
        });
    }

}

File: app/src/main/java/com/example/amazonprimeclone/fragments/home/series/SeriesFragmentViewModel.kt
package com.example.amazonprimeclone.fragments.home.series

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.amazonprimeclone.data.remote.Constants
import com.example.amazonprimeclone.modal.SeriesCategoryResults
import com.example.amazonprimeclone.modal.SeriesResponseResults
import com.example.amazonprimeclone.retrofit.repository.RetrofitMovieRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.launch
import javax.inject.Inject

@HiltViewModel
class SeriesFragmentViewModel @Inject constructor(
    private val retrofitMovieRepository: RetrofitMovieRepository
): ViewModel() {

    private val slideList: MutableLiveData<List<SeriesResponseResults>> = MutableLiveData()
    private val resultsList: MutableLiveData<List<SeriesCategoryResults>> = MutableLiveData()

    init {
        getData()
    }

    fun getResultsList(): MutableLiveData<List<SeriesCategoryResults>> {
        return resultsList
    }

    fun getSlideList(): MutableLiveData<List<SeriesResponseResults>> {
        return slideList
    }

    private fun getData() {
        loadSlideList()
        loadResultList()
    }

    private fun loadSlideList() = viewModelScope.launch{
        val response = retrofitMovieRepository.getTopRatedSeries()
        if(response!=null){
            slideList.postValue(response.results.subList(0,9))
        }
    }

    private fun loadResultList() = viewModelScope.launch{
        val response = retrofitMovieRepository.getCategoriesSeries(Constants.API_KEY)
        if(response!=null){
            resultsList.postValue(response.genres)
        }
    }

}